File: service/src/main/java/net/huray/backend/minuting/error/FeignErrorDecoder.java
package net.huray.backend.minuting.error;

import feign.Response;
import feign.RetryableException;
import feign.codec.ErrorDecoder;
import org.springframework.stereotype.Component;

@Component
public class FeignErrorDecoder implements ErrorDecoder {

    @Override
    public Exception decode(final String methodKey, Response response) {
        return new RetryableException(response.status(), response.reason(), response.request().httpMethod(), null, response.request());
    }

}

File: service/src/main/kotlin/net/huray/backend/minuting/support/FeignErrorDecoder.kt
package net.huray.backend.minuting.support

import feign.Response
import feign.RetryableException
import feign.codec.ErrorDecoder
import net.huray.backend.http.exception.BaseException
import org.springframework.stereotype.Component

@Component
class FeignErrorDecoder : ErrorDecoder {

    override fun decode(methodKey: String, response: Response): Exception = with(response) {
        if (status() >= 500) RetryableException(status(), reason(), request().httpMethod(), null, request())
        else BaseException(status(), reason())
    }

}